packages/provider/src/wallet.ts

```diff
@@ -248,15 +249,37 @@ export class Wallet implements WalletProvider {
     })
   }
 
-  loadSession = async (): Promise<WalletSession | undefined> => {
+  loadSession = async (preferredNetwork?: string | number): Promise<WalletSession | undefined> => {
     const data = await LocalStorage.getInstance().getItem('@sequence.session')
     if (!data || data === '') {
       return undefined
     }
+
     try {
       const session = JSON.parse(data) as WalletSession
       if (session) {
-        this.useSession(session, false)
+        // Setting preferredNetwork as default network if it's in session.networks
+        if (preferredNetwork !== undefined) {
+          const preferredNetworkIdNum = parseInt(preferredNetwork as any)
```

Comment:
Applying `parseInt` to non-strings is discouraged: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseInt#using_parseint_on_non-strings.